automations/actions.rst

```diff
@@ -963,37 +458,164 @@ Configuration variables:
 - **condition** (**Required**, :ref:`Condition <config-condition>`):
   The condition to check.
 
-.. _interval:
+.. _lambda_condition:
 
-``interval`` Component
-----------------------
+``lambda`` Condition
+********************
 
-This component allows you to run actions at fixed time intervals.
-For example if you want to toggle a switch every minute, you can use this component.
-Please note that it's possible to achieve the same thing with the :ref:`time.on_time <time-on_time>`
-trigger, but this technique is more light-weight and user-friendly.
+This condition performs an arbitrary piece of C++ code (see :ref:`Lambda <config-lambda>`)
+and can be used to create conditional flow in actions.
 
 .. code-block:: yaml
 
-    # Example configuration entry
-    interval:
-      - interval: 1min
-        then:
-          - switch.toggle: relay_1
+    on_...:
+      then:
+        - if:
+            condition:
+              # Should return either true or false
+              lambda: |-
+                return id(some_sensor).state < 30;
+            # ...
 
+All Triggers
```

Comment:
I think we should drop these blocks, they are missing a lot and not always kept up to date.

Reply:
There were several references to the `conditions` section -- removed in [`2dad574`](https://github.com/esphome/esphome-docs/pull/3957/commits/2dad574cf93d39db91e604ccd6c49d7965d1b409) in case you're curious what/where they were.

Reply:
Maybe link to `common_conditions`?

Reply:
Or perhaps `Conditions` (and Triggers) needs its own header (or page) where a general explanation of them plus the common ones can go.